### ISSUE TITLE
dns zone visibility not written to state in 2.4

### ISSUE THREAD

**Issue by dolvany** [terraform-providers/terraform-provider-google#3435]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version

Terraform v0.11.13
+ provider.google v2.4.0

### Affected Resource(s)

google_dns_managed_zone

### Steps to Reproduce

`terraform import`

### Expected Behavior

state contains `visibility="public"`

### Actual Behavior

state contains `visibility=""`

### Result

`visibility:     "" => "public" (forces new resource)`

### Workaround

`version = "= 2.3"`

**Comment by Starefossen** [comment]
Can confirm this bug in version 2.4 of the Google Provider.

**Comment by SGrondin** [comment]
This is also affecting us. We will not be able to upgrade from 2.3 until it's addressed.

**Comment by kornface13** [comment]
I'm also stuck on 2.3 due to this.

**Comment by pdecat** [comment]
Same issue here.

Also, while trying to work it around by adding the following:

```
  visibility = "public"

  private_visibility_config = {
    networks {}
  }
```

It crashed the provider:

```
Error: Error running plan: 1 error(s) occurred:

* module.global.module.dns.google_dns_managed_zone.objenio_us: 1 error(s) occurred:

* module.global.module.dns.google_dns_managed_zone.objenio_us: unexpected EOF


panic: interface conversion: interface {} is nil, not map[string]interface {}
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4:
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4: goroutine 386 [running]:
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4: github.com/terraform-providers/terraform-provider-google/google.resourceDn
sManagedZone.func1(0x0, 0x0, 0x1)
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4:        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-
providers/terraform-provider-google/google/resource_dns_managed_zone.go:78 +0x321
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4: github.com/terraform-providers/terraform-provider-google/vendor/github.com
/hashicorp/terraform/helper/schema.(*Set).hash(0xc00000bb20, 0x0, 0x0, 0xc000132820, 0xc000132820)
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4:        /opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform$
providers/terraform-provider-google/vendor/github.com/hashicorp/terraform/helper/schema/set.go:211 +0x3d
2019-04-16T17:51:51.616+0200 [DEBUG] plugin.terraform-provider-google_v2.4.0_x4: github.com/terraform-providers/terraform-provider-google/vendor/github.com
```

**Comment by pdecat** [comment]
A work around is to add `  visibility = ""`

**Comment by rileykarson** [comment]
Hey all! This is a known issue, thanks for your reports. Until we fix this, you can set visibility to `""` in order to suppress this diff, which should be the same as setting it to `"public"`. `"public"` _should_ be the default being returned, but I believe the API suppresses it in the response, returning no value if it would be `public`.

**Comment by brandoconnor** [comment]
Any chance we could get a patch release to fix this and any other minor issues in `v2.4.0`? The first item in the `v2.5.0` changelog has me a little wary that it'd be more of a major version rev rather than minor.

**Comment by paddycarver** [comment]
Hi @brandoconnor,

That's fair. I think we're currently hoping the 2.5.0 upgrade will be small enough that people will be able to do it, but in the event that people need the patch release and can't do 2.5.0 for whatever reason, I think we want to have a discussion about shipping a 2.4.1 after a 2.5.0. That's not something that has been done before, so I don't want to promise it, but it's possible in theory. Unfortunately, the state of our unreleased changes and the timeline on releasing a 2.5.0 is such that releasing a 2.4.1 is a tough proposition for us right now, but we're monitoring the situation and will change our stance as needed as events develop.

**Comment by Starefossen** [comment]
This has been fixed the latest version of the Google Cloud provider v1.5.1.

**Comment by rileykarson** [comment]
In addition to `2.5.1`, this has also been fixed in version `2.4.1` of the provider, released out of band as a backport in case the `0.12` integration causes issues for users.

**Comment by brandoconnor** [comment]
Many thanks!
